### ISSUE TITLE
Asynchronous Processing like f# Mailboxprocessor

### ISSUE THREAD

**Issue by furesoft** [rzubek/CSLisp#29]
An C# Implementation exits so we could use this

**Comment by rzubek** [comment]
I know nothing about that, but it sounds interesting.

Are you interested in implementing something like that?
Also, please provide more info.

**Comment by furesoft** [comment]
I can do it.

The concept is really easy. You put messages on a message queue and a message processor process one message at a time. so you prevent that a ressource is shared over threads.

**Comment by furesoft** [comment]
And all is asynchronous

**Comment by rzubek** [comment]
Sounds useful. Can you please post some links to more info about this? (And especially the existing C# implementation.)

**Comment by furesoft** [comment]
https://github.com/kthompson/Actress

https://www.codemag.com/Article/1707051/Writing-Concurrent-Programs-Using-F

**Comment by rzubek** [comment]
Thanks for these!

There's one constraint: one main design goal for CSLisp is _"Easy to embed and use in C# / .NET - no extra dependencies"_ (see Readme). The main dll only references .Net Standard and no other third-party code.

So if you're considering integrating an existing implementation, that wouldn't be a good fit for this project. (But you can fork it and add the integration to your version.)

**Comment by furesoft** [comment]
But we can write our own implementation, then its depdendency free or we add an extra package that can be included

**Comment by rzubek** [comment]
Yes, writing a new implementation that's free of dependencies would be great! 

Basically the goal is to avoid adding extra dependencies.
